posthog/api/action.py

```diff
@@ -304,28 +294,56 @@ def trends(self, request: request.Request, *args: Any, **kwargs: Any) -> Respons
         actions = actions.filter(deleted=False)
         actions_list = []
 
-        parsed_actions = self._parse_actions(request)
+        parsed_actions = self._parse_entities('actions')
+        parsed_events = self._parse_entities('events')
+        def _trend_entity_factory(id, name):
+            return {
+                'action': {
+                    'id': id,
+                    'name': name
+                },
+                'label': name,
+                'count': 0,
+                'breakdown': []
+            }
 
+        def _process_trend_entity(entity, filters={}, entity_type=None):
+            if entity_type == 'action':
+                trend_entity = _trend_entity_factory(entity.pk, entity.name)
+                filtered_events = Event.objects.query_db_by_action(entity)
```

Comment:
This should be `filter_by_action`, which uses the precalculated relational table, rather than doing a really heavy query. `query_db_by_action` is only used to generate that precalculated table.